src/main/java/com/mparticle/kits/AppboyKit.java

```diff
@@ -536,4 +559,84 @@ Calendar getCalendarMinusYears(int years) {
             return null;
         }
     }
+
+    abstract class StringTypeParser {
+
+        void parseValue(String key, String value) {
+            if (value.equalsIgnoreCase("true") || value.equalsIgnoreCase("false")) {
```

Comment:
safer to flip these? "true".equalsIgnoreCase

Reply:
good call. I also moved the catch block to encompass the Boolean conversion as well..figure better safe than sorry